src/js/login.js

```diff
@@ -11,6 +11,40 @@ lg.loadLogin = function () {
     });
 }
 
+lg.googleLogin = function (gToken) {
+    var cellUrl = sessionStorage.googleLoginCell;
+    $.ajax({
+        type: "POST",
+        url: cellUrl + "__token",
+        data: {
+            grant_type: 'urn:x-personium:oidc:google',
+            id_token: gToken
+        },
+        headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/x-www-form-urlencoded'
+        }
+    }).done(function (data) {
+        data.cellUrl = cellUrl;
+        var i = cellUrl.indexOf("/"); // first slash
+        i = cellUrl.indexOf("/", i + 2);  // second slash
+        data.baseUrl = cellUrl.substring(0, i + 1);
+        data.profile = JSON.parse(sessionStorage.getItem("myProfile"));
+        //data.userName = username;
+        data.userName = "googleAccount";
+        var pos = location.href.indexOf("id_token");
```

Comment:
should not use location anymore, why not use cellUrl?

Reply:
Since data.cellUrl is the URL of cell, it does not contain information after #. data.cellUrl does not include id_token.

Reply:
You are not using token here or after.

Reply:
I'm sorry. It was inappropriate.
Because we want to keep objects (mode, target) other than id_token as is, cellUrl can not be used.

Reply:
Now it is clear. But need to decide whether we still want to support mode in current design.